lib/Test2/Compare/Negatable.pm

```diff
@@ -0,0 +1,119 @@
+package Test2::Compare::Negatable;
+use strict;
+use warnings;
+
+require overload;
+require Test2::Util::HashBase;
+
+sub import {
+    my ($pkg, $file, $line) = caller;
+
+    my $sub = eval <<"    EOT" or die $@;
+package $pkg;
+#line $line "$file"
+sub { overload->import('!' => 'clone_negate', fallback => 1); Test2::Util::HashBase->import('negate')}
```

Comment:
It is necessary because compare objects are r-eusable. What if you make my $bcheck = bool(1); then later use it in one function is(0, !$bcheck); then another as is(1, $bcheck). Any sane person would expect both to pass, but if ! mutates $bcheck that expectation is broken. Having ! clone it and modify the copy is much more sane.

The reason I abstracted it out is because bool is only the latest negatable compare type, I wanted to consistently apply this behavior to the already existing negatable types.